src/zenml/cli/stack_recipes.py

```diff
@@ -867,33 +874,81 @@ def deploy(
                     stack_recipe_service = StackRecipeService(
                         config=terraform_config
                     )
+
                 # start the service (the init and apply operation)
                 stack_recipe_service.start()
 
+                # invoke server deploy
+                if no_server:
+                    logger.info(
+                        "The --no-server flag was passed. "
+                        "Skipping the remote deployment of ZenML. "
+                        "Please note that if you wish to use the stack "
+                        "that you created through this recipe, you will "
+                        "need to deploy ZenML on the cloud."
+                    )
+                else:
+                    if zen_server_exists():
+                        logger.info(
+                            "A ZenML deployment exists already with URL: "
+                            f"{GlobalConfiguration().store.url}. The recipe will "
+                            "not create a new installation."
+                        )
+                    else:
+                        logger.info("No remote deployment of ZenML detected. ")
+                        vars = stack_recipe_service.get_vars()
+                        filter = [
+                            "aws-stores-minimal",
+                            "azureml-minimal",
+                            "vertex-ai",
+                        ]
+                        if Path(
+                            stack_recipe_service.terraform_client.working_dir
+                        ).name in filter and (
+                            "enable_mlflow" not in vars
```

Comment:
I'm not sure I understand why we care if MLflow is enabled/disabled here?

Reply:
Only if mlflow is enabled on the recipe side, a Kubernetes cluster gets created to host it. Otherwise, no cluster and therefore we can't do a `zenml deploy` right away.